### ISSUE TITLE
Messed up user agent 

### ISSUE THREAD

**Issue by Dani-1-2** [arkenfox/user.js#1787]
I am using firefox 121.0 on macOS 14.1 Sonoma. The issue is that when I open https://amiunique.org/fingerprint, the http header attribute shows a windows user agent ( Mozilla/5.0 (Windows NT 10.0; rv:121.0) Gecko/20100101 Firefox/121.0), but the  javascript attribute shows my real user agent (Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:121.0) Gecko/20100101 Firefox/121.0) and the platform as MacIntel. I expected it to change both use agent attributes. How do I prevent this from happening? I am using a slightly modified user.js file, but it does not change anything related to the user agent.
Console warnings:
Loading failed for the <script> with source “https://amiunique.org/analytics.js”. [amiunique.org:564:168](https://amiunique.org/)
Blocked https://amiunique.org/fingerprint from extracting canvas data because no user input was detected. [198df04.js:2:18344](https://amiunique.org/_nuxt/198df04.js)
Request for font "Avenir Next Condensed" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Herculanum" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Hiragino Kaku Gothic Pro" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Hiragino Kaku Gothic ProN" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Hiragino Kaku Gothic Std" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Hiragino Kaku Gothic StdN" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Hiragino Maru Gothic Pro" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Hiragino Mincho Pro" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Marion" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Request for font "Nadeem" blocked at visibility level 1 (requires 3)
[198df04.js:2:20033](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43164](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43187](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43164](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43187](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43164](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43187](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled. [198df04.js:2:43164](https://amiunique.org/_nuxt/198df04.js)
Failed to create WebGL context: WebGL is currently disabled.

**Comment by rusty-snake** [comment]
>  the http header attribute shows a windows user agent, but the javascript attribute shows my real user agent and the platform as MacIntel. I expected it to change both use agent attributes.

This is intended behaviour 

https://github.com/arkenfox/user.js/blob/4a510a4b4ceda8a177831a77309f2faccf7e18d5/user.js#L701-L703

> real user agent and the platform as MacIntel.

It's not your real UA, it is still spoofed (Mac OS X 10.15 vs. macOS 14.1).

Also, platform can not be hidden.

> How do I prevent this from happening? 

TL;DR: Don't.

Long: Do not mess with UA, either use RFP as is or disable RFP and go with your real UA.

> slightly modified user.js file

Then post also your modifications.

**Comment by Dani-1-2** [comment]
> > the http header attribute shows a windows user agent, but the javascript attribute shows my real user agent and the platform as MacIntel. I expected it to change both use agent attributes.
> 
> This is intended behaviour
> 
> https://github.com/arkenfox/user.js/blob/4a510a4b4ceda8a177831a77309f2faccf7e18d5/user.js#L701-L703
> 
> > real user agent and the platform as MacIntel.
> 
> It's not your real UA, it is still spoofed (Mac OS X 10.15 vs. macOS 14.1).
> 
> Also, platform can not be hidden.
> 
> > How do I prevent this from happening?
> 
> TL;DR: Don't.
> 
> Long: Do not mess with UA, either use RFP as is or disable RFP and go with your real UA.
> 
> > slightly modified user.js file
> 
> Then post also your modifications.

It always dispays macOS 10.15, even without a modified user.js, so this is not spoofed. Here is the modified user.js that I use: https://github.com/Dani-1-2/user.js. It doesn't delete the website data on close, removes pocket and disables letterboxing.

**Comment by Thorin-Oakenpants** [comment]
the version is but one part of a userAgent

```
	/*
	windows:
	- FF88+ 1693295: capped at 10.0
	- FF116+ 1841425: windows hardcoded to 10.0 (patched 117 but 115 was last version for < win10)
	mac:
	- FF87+ 1679929: capped at 10.15
	- FF116+ 1841215: mac hardcoded to 10.15 (patched 117 but 115 was last release for < 10.15)
	android:
	- FF122+ 1865766: android hardcoded to 10.0
	*/
```

RFP still protects (by standardizing per OS) other values in navigator and within the actual userAgent
